### PR TITLE
item prices: Show alch price while alching

### PR DESCRIPTION
**Pull request by Hydrox6**
I always found it weird that you couldn't see the value of the item you were about to alch, so here it is

Also works with the Explorer's Ring interface!

### CODE REVIEW

**Comment by deathbeam** [comment]
This should be configurable. Also needs rebase

**Comment by Hydrox6** [comment]
Rebased, added configuration

File: runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesOverlay.java [hunk 1]
```diff
@@ -96,6 +97,11 @@ public Dimension render(Graphics2D graphics)
 		// Tooltip action type handling
 		switch (action)
 		{
+			case ITEM_USE_ON_WIDGET:
+				if (!menuEntry.getOption().equals("Cast") || !menuEntry.getTarget().contains("High Level Alchemy") || !config.showWhileAlching())
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesOverlay.java]: move the boolean check at start of if

File: runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesConfig.java [hunk 2]
```diff
@@ -86,4 +86,14 @@ default boolean showAlchProfit()
 		return false;
 	}
 
+	@ConfigItem(
+			keyName = "showWhileAlching",
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesConfig.java]: Fix your indentation

**Comment by Hydrox6** [comment]
Rebased and updated to work with #9070

File: runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesOverlay.java [hunk 3]
```diff
@@ -95,9 +95,9 @@ public Dimension render(Graphics2D graphics)
 		switch (action)
 		{
 			case ITEM_USE_ON_WIDGET:
-				if (!menuEntry.getTarget().contains("High Level Alchemy") || !config.showAlchProfit())
+				if (!config.showWhileAlching() || !menuEntry.getOption().equals("Cast") || !menuEntry.getTarget().contains("High Level Alchemy"))
 				{
-					break;
+					return null;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesOverlay.java]: There is no reason to change this to return null when you can just use break like it is atm on master

**Pull request merged**